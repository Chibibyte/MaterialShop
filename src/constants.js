export default {
    BUTTONICONS: {
        AUTHENTICATION: 'mdi-login-variant',
        ADDRESS: 'mdi-package-variant',
        BANKACCOUNT: 'mdi-bank',
        SHIPPING: 'mdi-truck',
        PAYMENT: 'mdi-cash',
        SUMMARY: 'mdi-text-box-check-outline',
        CHECKOUT_COMPLETE: 'mdi-check',
        ADD: 'mdi-plus',
        REMOVE: 'mdi-minus',
        CANCEL: 'mdi-window-close',
        CONFIRM: 'mdi-check',
        SHOPPING: 'mdi-shopping',
        SUCCESS: 'mdi-check',
        FAILURE: 'mdi-window-close',
        WARNING: 'mdi-alert-circle-outline'
    }
}