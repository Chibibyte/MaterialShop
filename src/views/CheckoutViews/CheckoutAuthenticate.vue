<template>
  <v-container class="d-flex green justify-center">
    <v-form class="white pa-5" ref="form" v-model="valid" lazy-validation>
      <v-text-field ref="emailInput" v-model="email" :rules="emailRules" label="E-mail" required></v-text-field>

      <v-text-field
        ref="passwordInput"
        v-model="password"
        :rules="passwordRules"
        label="Password"
        type="password"
        required
      ></v-text-field>

      <v-btn :disabled="!valid" color="success" class="mr-4" @click="validate">next</v-btn>
    </v-form>
  </v-container>
</template>


<script>
/**
 * Business Logic
 *    -display authentication form (like login form)
 *    -when user clicks on next
 *        -run validate()
 *
 *
 * Inputs: none
 * Outputs:
 *    -emailRules
 *    -passwordRules
 */
export default {
  name: "CheckoutAuthenticate",
  data: () => ({
    valid: true,
    email: "ro@bo.cop",
    emailRules: [
      v => !!v || "E-mail is required",
      v => /.+@.+\..+/.test(v) || "E-mail must be valid",
      v => v === "ro@bo.cop" || 'E-mail must be "ro@bo.cop"'
    ],
    password: "1234",
    passwordRules: [
      v => !!v || "Password is required",
      v => (v && v === "1234") || "Password must be 1234"
    ]
  }),
  methods: {
    validate() {
      this.$refs.form.validate();
      if (this.valid) {
        this.$store.commit("checkout/setCheckoutState", {
          chState: "authentication",
          value: true
        });
        this.$router.push("address");
      }
    }
  }
};
</script>