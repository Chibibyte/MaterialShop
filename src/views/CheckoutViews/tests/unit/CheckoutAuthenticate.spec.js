
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutAuthenticate from '@/views/CheckoutViews/CheckoutAuthenticate.vue';
import constants from '@/constants.js'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.prototype.$constants = constants;

let store;
let state;
let modules;
let mocks;
let stubs = [
    'v-container',
    'v-form',
    'v-text-field',
    'v-btn',
]
let wrapper;

beforeEach(() => {
    state = {}
    modules = {
        noncheckout: {
            namespaced: true,
            state() {
                return {
                }
            }
        },
        checkout: {
            namespaced: true,
            state() {
                return {
                    checkoutStates: {
                        authentication: false,
                        deliveryAddress: false,
                        bankaccount: false,
                        shipping: false,
                        payment: false,
                        summary: false,
                    }
                }
            },
            mutations: {
                setCheckoutState: jest.fn()
            }
        }
    }
    mocks = {
        $router: { push: jest.fn() }
    };
    store = new Vuex.Store({
        state,
        modules
    })
    wrapper = shallowMount(CheckoutAuthenticate, { mocks, stubs, store, localVue });
    wrapper.vm.$refs.form.validate = jest.fn();
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
});

describe('validation', () => {
    test('validate', () => {
        wrapper.vm.validate();
        expect(wrapper.vm.$refs.form.validate.mock.calls).toHaveLength(1);
    })

    // email
    test('empty email', () => {
        expect(wrapper.vm.emailRules[0]('')).toBe('E-mail is required');
        expect(wrapper.vm.emailRules[1]('')).toBe('E-mail must be valid');
        expect(wrapper.vm.emailRules[2]('')).toBe('E-mail must be "ro@bo.cop"');
    })

    test('wrong email value', () => {
        expect(wrapper.vm.emailRules[0]('ro@bo.co')).toBe(true);
        expect(wrapper.vm.emailRules[1]('ro@bo.co')).toBe(true);
        expect(wrapper.vm.emailRules[2]('ro@bo.co')).toBe('E-mail must be "ro@bo.cop"');
    })

    test('wrong email format', () => {
        expect(wrapper.vm.emailRules[0]('ro.bo.co')).toBe(true);
        expect(wrapper.vm.emailRules[1]('ro.bo.co')).toBe('E-mail must be valid');
        expect(wrapper.vm.emailRules[2]('ro.bo.co')).toBe('E-mail must be "ro@bo.cop"');
    })

    // password
    test('empty password', () => {
        expect(wrapper.vm.passwordRules[0]('')).toBe('Password is required');
        expect(wrapper.vm.passwordRules[1]('')).toBe('Password must be 1234');
    })

    test('wrong password value', () => {
        expect(wrapper.vm.passwordRules[0]('ro@bo.co')).toBe(true);
        expect(wrapper.vm.passwordRules[1]('ro@bo.co')).toBe('Password must be 1234');
    })
})



