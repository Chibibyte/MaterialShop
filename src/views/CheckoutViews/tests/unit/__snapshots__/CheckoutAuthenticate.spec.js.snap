// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<v-container-stub
  class="d-flex green justify-center"
>
  <v-form-stub
    class="white pa-5"
    lazy-validation=""
    value="true"
  >
    <v-text-field-stub
      label="E-mail"
      required=""
      rules="v => !!v || \\"E-mail is required\\",v => /.+@.+\\\\..+/.test(v) || \\"E-mail must be valid\\",v => v === \\"ro@bo.cop\\" || 'E-mail must be \\"ro@bo.cop\\"'"
      value="ro@bo.cop"
    />
     
    <v-text-field-stub
      label="Password"
      required=""
      rules="v => !!v || \\"Password is required\\",v => v && v === \\"1234\\" || \\"Password must be 1234\\""
      type="password"
      value="1234"
    />
     
    <v-btn-stub
      class="mr-4"
      color="success"
    >
      next
    </v-btn-stub>
  </v-form-stub>
</v-container-stub>
`;
